bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+the signature to put in the `channel_announcement_v2` message. Unlike the legacy
+`announcement_signatures` message, `announcement_signatures_v2` only has one
+signature field. This field is a MuSig2 partial signature which is the
+aggregation of the two signatures that the sender would have created (one for
+`bitcoin_key_x` and another for `node_ID_x`).
+
+## The `channel_announcement_2` Message
+
+This gossip message contains ownership information regarding a taproot channel. 
+It ties each on-chain Bitcoin key that makes up the taproot output key to the 
+associated Lightning node key, and vice-versa. The channel is not practically 
+usable until at least one side has announced its fee levels and expiry, using 
+`channel_update_2`.
+
+See [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification)
+for more information regarding the requirements for proving the existence of a
+channel.
+
+1. type: xxx (`channel_announcement_2`)
+2. data: (tlv_stream):
+   1. type: 0 (`announcement_sig`)
+   2. data:
+        * [`schnorr_signature`:`schnorr_signature`]
+   3. type: 1 (`features`)
+   4. data:
+        * [`...*byte`: `features`]
+   5. type: 2 (`chain_hash`)
+   6. data:
+        * [`chain_hash`:`chain_hash`]
+   7. type: 3 (`short_channel_id`)
+   8. data:
+        * [`short_channel_id`:`short_channel_id`]
+   9. type: 4 (`node_id_1`)
+   10. data:
+        * [`point`:`point`]
+   11. type: 5 (`node_id_2`)
+   12. data:
+       * [`point`:`point`]
+   13. type: 6 (`bitcoin_key_1`)
+   14. data:
+       * [`point`:`point`]
+   15. type: 7 (`bitcoin_key_2`)
+   16. data:
+       * [`point`:`point`]
+
+### Requirements
+
+The origin node:
+
+- MUST set `chain_hash` to the 32-byte hash that uniquely identifies the chain
+  that the channel was opened within:
+    - for the _Bitcoin blockchain_:
+        - MUST set `chain_hash` value (encoded in hex) equal
+          to `6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000`.
+- MUST set `short_channel_id` to refer to the confirmed funding transaction, as
+  specified in [BOLT #2](02-peer-protocol.md#the-channel_ready-message).
+    - Note: the corresponding output MUST be a P2TR, as described
+      in [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification).
+- MUST set `node_id_1` and `node_id_2` to the public keys of the two nodes
+  operating the channel, such that `node_id_1` is the lexicographically-lesser
+  of the two compressed keys sorted in ascending lexicographic order.
+- MUST set `bitcoin_key_1` and `bitcoin_key_2` to `node_id_1` and `node_id_2`'s
+  respective `funding_pubkey`s.
+- MUST set `features` based on what features were negotiated for this channel,
+  according to [BOLT #9](09-features.md#assigned-features-flags)
+- MUST set `schnorr_signature` to a valid signatures of the message `m` (the
+  double-SHA256 hash of the serialised message excluding the `announcement_sig`
```

Comment:
double-SHA256?  That seems a bit unnatural in this context?

Please use BIP-340's recommendation here.  In particular, as bolt12 does:

```
All signatures are created as per
[BIP-340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki)
and tagged as recommended there.  Thus we define H(`tag`,`msg`) as
SHA256(SHA256(`tag`) || SHA256(`tag`) || `msg`), and SIG(`tag`,`msg`,`key`)
as the signature of H(`tag`,`msg`) using `key`.

Each form is signed using one or more *signature TLV elements*: TLV
types 240 through 1000 (inclusive).  For these,
the tag is "lightning" || `messagename` || `fieldname`.
```

That makes the tag "lightningchannel_announcement_2schnorr_signature".  Now, we use the merkle as the message in bolt12, which is overkill: SHA256(other fields) should suffice?